### DISCUSSION TITLE
Feature request: Shell completions

### DISCUSSION THREAD

**Discussion by Laura7089** [nextest-rs/nextest#1755] [Feature requests]
It would be convenient for users if the `cargo nextest` interface provided shell completions for major shells.

**Comment by sunshowers** [comment]
It certainly would be, agreed. Unfortunately that would require several things:

1. To write completion scripts, or use the in-progress clap dynamic completion work.
2. For Cargo's own completion script to be able to delegate to subcommands, something that isn't possible at the moment.

Someone would have to drive both of these projects. Realistically speaking I don't anticipate getting to this any time within at least the next 12 months.
